server/controller/user.controller.js

```diff
@@ -35,274 +34,233 @@ import { User } from '../database/models';
-          delete userToReturn.hashedPassword;
-          res.status(200).json(userToReturn);
-        } else {
-          generateServerErrorCode(
+      await User.findOne({ email }).then((err, user) => {
+        if (user && user.email) {
+          const isPasswordMatched = user.comparePassword(password);
+          if (isPasswordMatched) {
+            // Sign token
+            const token = jwt.sign({ email }, config.passport.secret, {
+              expiresIn: 1000000,
+            });
+            const userToReturn = { ...user.toJSON(), ...{ token } };
+            delete userToReturn.hashedPassword;
+            return res.status(200).json(userToReturn);
+          }
+          return generateServerErrorCode(
             res,
             403,
             'login password error',
             WRONG_PASSWORD,
             'password'
           );
         }
-      } else {
-        generateServerErrorCode(
+        return generateServerErrorCode(
           res,
           404,
           'login email error',
           USER_DOES_NOT_EXIST,
           'email'
         );
-      }
+      });
     } catch (e) {
-      generateServerErrorCode(res, 500, e, SOME_THING_WENT_WRONG);
+      return generateServerErrorCode(res, 500, e, SOME_THING_WENT_WRONG);
     }
-  }
+  });
 });
 
 // @route   PUT /profile
 // @desc    Edits a user's profile
 // @access  Private
 userController.put(
-  '/profile/:userId',
+  '/profile',
   passport.authenticate('jwt', { session: false }),
   validateEditProfile,
-  async (req, res) => {
-    // Check if the validators caught any invalid inputs
-    const errors = validationResult(req);
-    if (errors.isEmpty() === false) {
-      return res.status(400).json(errors);
-    }
-
-    const { userId } = req.params;
-    const {
-      avatarUrl,
-      avatarType,
-      firstName,
-      lastName,
-      bio,
-      major,
-      minor,
-      catalogYear,
-      gradDate,
-    } = req.body;
+  (req, res) => {
+    validationHandler(req, res, async () => {
+      const { user } = req;
+      const {
+        avatarUrl,
+        avatarType,
+        firstName,
+        lastName,
+        bio,
+        major,
+        minor,
+        catalogYear,
+        gradDate,
+      } = req.body;
 
-    // Check if the client had included at least one data to be changed
-    if (
-      !avatarUrl &&
-      !avatarType &&
-      !firstName &&
-      !lastName &&
-      !bio &&
-      !major &&
-      !minor &&
-      !catalogYear &&
-      !gradDate
-    ) {
-      // Client did not include data to be changed so there is nothing to be updated, return error code
-      return res.status(400).json({ errors: PARAMETERS_REQUIRED });
-    }
-
-    // Check if the client had inputted gradDate and check if it's a valid object input
-    if (gradDate && checkIfCorrectGradDate(gradDate) === false) {
-      return res.status(400).json({ errors: INVALID_GRAD_DATE });
-    }
+      if (
```

Comment:
More of a comment for @DaleCS, could this validation be done in the validateEditProfile instead of here?